templates/environment/partials/vpc-endpoints.yml

```diff
@@ -0,0 +1,83 @@
+VPCEndpointSecurityGroup:
+  Condition: CreateVPCEndpoints
+  Metadata:
+    'aws:copilot:description': 'A security group to associate with your VPC endpoints'
+  Type: AWS::EC2::SecurityGroup
+  Properties:
+    GroupDescription: Security Group to allow use of VPC Endpoints.
+    VpcId: !Ref VPC
+    SecurityGroupIngress:
+      - IpProtocol: tcp
+        Description: HTTPS
+        FromPort: 443
+        ToPort: 443
+        CidrIp: {{.CIDR}}
+
+# Allow ingress traffic from VPC endpoints to our containers.
+IngressFromVPCEndpoints:
+  Condition: CreateVPCEndpoints
+  Type: AWS::EC2::SecurityGroupIngress
+  Properties:
+    Description: Ingress from VPC endpoints.
+    GroupId: !Ref 'EnvironmentSecurityGroup'
+    IpProtocol: -1
+    SourceSecurityGroupId: !Ref 'VPCEndpointSecurityGroup'
+
+# Gateway endpoint for S3.
```

Comment:
We might want to put it in the metadata description for those endpoints so that it can be tracked by our new progress tracker (copy @efekarakus)?